src/main/java/org/mule/extension/db/internal/domain/connection/DbConnection.java

```diff
@@ -136,6 +138,31 @@ default Struct createStruct(String typeName, Object[] values) throws SQLExceptio
     return getJdbcConnection().createStruct(typeName, values);
   }
 
+
+  /**
+   * Retrieves a description of the given stored procedure parameter and result columns. For more information
+   * {@see DatabaseMetaData#getProcedureColumns}
+   * </p>
+   * Some Databases assign a different meaning to each parameter. You should override this method when needed.
+   *
+   * @param storedProcedureName the stored procedure name
+   * @param storedProcedureOwner the owner of the stored procedure
+   * @param storedProcedureParentOwner the owner of the owner of the stored procedure
+   * @param catalogName the catalog name where the stored procedure is defined
+   * @return <code>ResultSet</code> - each row describes a stored procedure parameter or column
+   * @throws SQLException if a database access error occurs
+   */
+  default ResultSet getProcedureColumns(String storedProcedureName, String storedProcedureOwner,
```

Comment:
Maybe have no default and implement it in DefaultConnection and implement it in XADbConnection?